### DISCUSSION TITLE
Import from VPSC

### DISCUSSION THREAD

**Discussion by waka25** [mtex-toolbox/mtex#1939] [Ask Anything] [answered]
Hi,

I am working with mtex 5.8.2 and want to plot pole figures from texture files generated using VPSC code.
I tried to follow the example on this page: [https://mtex-toolbox.github.io/VPSCImport.html]
but was not successful.

This is the code I used to create the odf :
```
cs = crystalSymmetry('cubic');

path2file = [filesep 'Users/wakibsaid/Documents/VPSC/VPSC_code-main/ex01_elast/FCC_roll'];
odf = ODF.load([path2file filesep 'TEX_PH1.OUT'],'CS',cs,'HalfWidth',10*degree);
```
I end up with the following errors:
<img width="1084" alt="Screenshot 2023-09-01 at 14 43 24" src="https://github.com/mtex-toolbox/mtex/assets/143181547/1170d465-5604-46d1-8a2c-bade62f3b675">

I checked the functions and it looked like it was stumbling on the following function with the error stated to be on the last line:

```
function [odf,interface,options] = load(fname,varargin)
% import ebsd data 
%
% Description
% *loadODF* is a high level method for importing ODF data from external
% files. It autodetects the format of the file. As parameters the method
% requires a filename and the crystal and specimen @symmetry. Furthermore,
% you can specify a comment to be associated with the data. In the case of
% generic ascii files each of which consist of a table containing in each
% row the euler angles of a certain orientation see
% <loadODF_generic.html loadODF_generic> for additional options.
%
% Syntax
%   odf = ODF.load(fname,cs,ss)
%
% Input
%  fname     - filename
%  cs, ss    - crystal, specimen @symmetry (optional)
%
% Options
%  interface  - specific interface to be used
%  comment    - comment to be associated with the data
%
% Output
%  odf - @ODF
%
% See also
% ImportEBSDData EBSD/calcODF ebsd_demo loadEBSD_generic

%  determine interface 
if ~check_option(varargin,'interface')
  [interface,options] = check_interfaces(fname,'ODF',varargin{:});
else
  interface = get_option(varargin,'interface');
  options = delete_option(varargin,'interface',1);
end

odf = feval(['loadODF_',char(interface)],fname,options{:});  
```
by luck I left the file name empty in my code while trying different inputs (as just apostrophes) and I was prompted to choose the file manually from a window, followed by an import wizard : 
<img width="492" alt="Screenshot 2023-09-01 at 14 54 04" src="https://github.com/mtex-toolbox/mtex/assets/143181547/4399a140-eccc-47e3-8fcf-c7651dbf7e46">

 and then a second window asking me to select the file again, and then it works.
I can also skip the import wizard by editing my code as such: 
`odf = ODF.load([path2file filesep ''],'CS',cs,'HalfWidth',10*degree,'Header',4,'ZXZ, Bunge','Columns',[1,2,3,4],'ColumnNames',{'Phi1' 'phi' 'Phi2' 'weight'});`

but I still need to select the file twice.

This is the texture file:
[TEX_PH1.OUT.zip](https://github.com/mtex-toolbox/mtex/files/12498641/TEX_PH1.OUT.zip)

#1908 From this previous query I can see that it does work for others using 5.8.2 so it may be something specific to my MacBook ? I can also update my version of mtex and try that instead since this seems to work.

Hope this was clear.
Thanks in advance.

Wakib

**Comment by kilir** [comment, marked as answer]
Hi,
did you try the solution provided here https://github.com/mtex-toolbox/mtex/discussions/1908 ?
Cheers,
Rüdiger

**Reply by waka25** [reply]
Hi,
Yes,  I have updated to 5.10 and tried the solution in #1908, it is working :).
I was just curious as to why my version of 5.8.2 wasn't working.
Thank you.
Wakib

**Reply by ralfHielscher** [reply]
With MTEX 5.10.1 everything should now work out of the box.

Ralf.
